### ISSUE TITLE
PAT RUNS FOR EVER 

### ISSUE THREAD

**Issue by rHorsey** [NREL/OpenStudio-server#252]
_From @tabuimara on July 26, 2017 15:22_

SINCE I UPDATED OPEN-STUDIO to 2.2.0, I am not able to perform any parametric analysis , the process starts and keeps in progress forever . I thought might be an issue with one of the weather files , so i tested them one at a time , nothing changes !

_Copied from original issue: NREL/OpenStudio#2701_

**Comment by rHorsey** [comment]
_From @DavidGoldwasser on July 26, 2017 19:42_

@tabuimara What is the status in the green bar? Does it show "Analysis Started" or is it something else? Also have you seen this on multiple PAT projects and does it still happen after a fresh restart of your computer.

**Comment by rHorsey** [comment]
_From @tabuimara on July 26, 2017 19:43_

Hi
It does start and says analysis starts

Tareq Isamil Abuimara, MSc, MRAIC
Cell: (613)898-1406
PhD Student/ TA/RA
Faculty of Engineering and Design

**Comment by rHorsey** [comment]
_From @DavidGoldwasser on July 26, 2017 20:37_

Assuming you are running on local machine, what do you see at http://localhost:8080/

Do you see an analysis, and does that analysis seem to have any datapoints, and in the GUI on the Run tab under the row for each Datapoint is the status "Started" for at least some of the datapoints?

**Comment by rHorsey** [comment]
_From @tabuimara on July 27, 2017 14:38_

Hi David
Sorry for my late reply.
Yes I am running on local server and the analysis starts, the data points are created and the stutus is started.
What happens is different each an every time , it either starts with no end or fails right away .
Thanks

Tareq Isamil Abuimara, MSc, MRAIC
Cell: (613)898-1406
PhD Student/ TA/RA
Faculty of Engineering and Design

**Comment by rHorsey** [comment]
_From @DavidGoldwasser on July 27, 2017 16:39_

Sorry, I have a few more questions. What operating system are you on, and if you restart your machine, make a fresh PAT project (something really basic without all of the same measures) does it run? If it is unique to a specific project might have you send it to us at OpenStudio@nrel.gov.

**Comment by rHorsey** [comment]
_From @tabuimara on July 27, 2017 16:43_

Thanks for your response
I am using Windows 64bit
I did the following and did not work :

  *   Restarted my computer several time
  *   Uninstall OS and Reinstall multiple times
  *   Changed the set of measures
I am now back to version 2.1.0 to see if the problem is with OS or my local computer , will get back to you as soon as I am done.
Thanks

Tareq Isamil Abuimara, MSc, MRAIC
Cell: (613)898-1406
PhD Student/ TA/RA
Faculty of Engineering and Design

**Comment by rHorsey** [comment]
_From @DavidGoldwasser on July 28, 2017 16:15_

@tabulimara. I got your email, and saw that now, you are not getting it running forever, but the datapoints are failing. The behavior appears to be related to issue #2557 which is closed and should be fixed, so I'm not sure why you are seeing it. If you are using a PAT project that pre-dates the fix, then I can see this happening. It sounds like this project could have bene made in 2.1.0 which didn't have the fix)

I'm seeing this error
```
	bin/delayed_job:6:in `<main>'
[11:52:14.211511 ERROR] C:/openstudio-2.2.0/pat/OpenStudio-server/gems/gems/openstudio-
workflow-1.3.1/lib/openstudio/workflow/util/measure.rb failed with message Error assigning 
argument in measure C:/openstudio-2.2.0/pat/OpenStudio-server/gems/gems/openstudio-
workflow-1.3.1/lib/openstudio/workflow/util/measure.rb. Failed with Could not set argument 
'construction' to value '', C:/openstudio-2.2.0/pat/OpenStudio-server/gems/gems/openstudio-
workflow-1.3.1/lib/openstudio/workflow/util/measure.rb:186:in `apply_arguments_2'
```
That appears to come from the Add Overhang by Projection Factor measure

```
    #make an argument for construction
    construction = OpenStudio::Ruleset::OSArgument::makeChoiceArgument("construction", construction_handles, construction_display_names,false)
    construction.setDisplayName("Optionally Choose a Construction for the Overhangs.")
    args << construction
```

I just got this information out of the run.log by unizpping one of the data_point.zip files in the localResults folder.  I haven't had a chance to run the project here, but What I would try is to remove the measure from the workflow (using the PAT application), save PAT, then re-add the measure, and see if this goes away. You can also just pick an opaque construction for the shading surfaces to see if that resolves the issue. You may hit similar issue in other measures for example the optional plenum zones. If I get a chance I'll try this approach on your project, but if you have a chance you may want to try it first.

**Comment by rHorsey** [comment]
_From @tabuimara on July 31, 2017 20:23_

Hi David
Hope you are doing well
I still have the same issue withPAT
DATAPOINT FAILURE
I have sent you the project earlier on the other email, any thoughts ?
Thanks


Tareq I. A. Abuimara
Get Outlook<https://aka.ms/qtex0l> for iOS
________________________________

**Comment by rHorsey** [comment]
_From @asparke2 on August 22, 2017 16:43_

@chrisbalbach reported this issue, demonstrated it, and I have a PAT project that replicates the issue (on Windows at least).  Basically what's happening is that one of the design alternatives is getting re-run over and over again and never completes, even though the simulation finishes and everything.

**Comment by rHorsey** [comment]
_From @mbcoalson on August 30, 2017 14:57_

I seem to be having a similar issue. I had difficulty getting PAT to run locally, so I uninstalled and reinstalled it directly in the C-drive. Changing where I have PAT installed has allowed me to connect to the local-virtual server. Unfortunately, I am receiving a datapoint failure status message on all my simulations, even as I have progressively simplified them (as is talked about in this thread). 

*I'm running Windows 10, on a 64bit system.* 

Here is the zip file associated with my most recent simulation.
[UCF_SU_2017.8.30_1.0.zip](https://github.com/NREL/OpenStudio/files/1264047/UCF_SU_2017.8.30_1.0.zip)

Any thoughts are greatly appreciated. 
Mathew Coalson
matt.coalson@gmail.com
mcoalson@hanson-inc.com

**Comment by rHorsey** [comment]
_From @asparke2 on September 16, 2017 22:46_

We believe that this issue is the result of having some reserved characters (periods, dollar signs) in the measure option and/or design alternative names.  This hypothesis is being tested and if it is true we will add GUI logic into PAT to warn users ahead of time.  The better long term solution to allow these characters in the names may be pursued in the future.

**Comment by DavidGoldwasser** [comment]
Confirmed it prevents me from running with an informative message.
![screen shot 2018-02-12 at 3 48 02 pm](https://user-images.githubusercontent.com/4857055/36124114-236e1792-100c-11e8-8c8e-78d8f202fe27.png)

**Comment by DavidGoldwasser** [comment]
Chris B. was able to reproduce this problem with arguments keys that contain periods (90.1). 
[03_LearningCenter.zip](https://github.com/NREL/OpenStudio-server/files/2358870/03_LearningCenter.zip)

**Comment by rHorsey** [comment]
The behavior gets funky starting with an exception [here](https://github.com/NREL/OpenStudio-server/blob/develop/server/app/workers/run_simulate_data_point.rb#L225)

**Comment by DavidGoldwasser** [comment]
I was able to resolve the issue for specific project by updating the measure to cleans names used similar to how runner.registerValues from model objects are cleansed in reporting measures. Arguments didn't cleanly update in PAT without deleting and re-adding the measure. I first tried changing the model used for arguments, and then changing it back. Seems like an easy fix could be to update PAT to catch this in pre-flight similar to periods in design alternative names.
